### PR TITLE
task solution

### PR DESCRIPTION
**Pull request by no4kar**
without nested loop

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,37 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const removeKeyFromStateObj = (key) => {
+    delete state[key];
+  };
+
+  for (const action of actions) {
+    if (Object.hasOwnProperty.call(action, 'type')) {
```
Comment (Esceype) [inline src/transformState.js]: Remove this as you have default case

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,37 @@
...
+      switch (action.type) {
+        case 'addProperties':
+          Object.keys(action.extraData).forEach(
+            (key) => {
+              state[key] = action.extraData[key];
+            });
+          continue;
```
Comment (Esceype) [inline src/transformState.js]: Why do you need `continue` here and in other cases ?

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,37 @@
...
+  };
+
+  for (const action of actions) {
+    if (Object.hasOwnProperty.call(action, 'type')) {
+      switch (action.type) {
+        case 'addProperties':
+          Object.keys(action.extraData).forEach(
+            (key) => {
+              state[key] = action.extraData[key];
+            });
```
Comment (Esceype) [inline src/transformState.js]: Rewrite it using Object.assign

**Review by Esceype — CHANGES_REQUESTED** [review]
Great job!!! Read the following comments bellow and let`s improve it a little bit

File: src/transformState.js (hunk 1 above)
Comment (no4kar) [reply src/transformState.js]: how can I be sure that this key('type') exist in nested object?

File: src/transformState.js (hunk 3 above)
Comment (no4kar) [reply src/transformState.js]: thank you. with this looks better.

**Review by TarasHoliuk — APPROVED** [review]
:+1:

**Pull request closed without merging**